vignettes/sablefish_example.Rmd

```diff
@@ -0,0 +1,175 @@
+
+```{r weight-length-data, message = FALSE, warning = FALSE}
+# Pull survey data 
+bds_survey <- nwfscSurvey::pull_bio(
+  common_name = common_name,
+  survey = "NWFSC.Combo"
+)
+# Estimate weight-length relationship by sex
+weight_length_estimates <- nwfscSurvey::estimate_weight_length(
+  bds_survey,
+  verbose = FALSE
+)
+```
+
+```{r print-weight-length-table}
+knitr::kable(weight_length_estimates, "markdown")
+```
+
+## Catch data
+
+The catch data must be summarized because it will be used in the second stage expansion of the biological data. Where, `formatCatch()` takes a long data frame as input and converts it to a wide data frame with one row per year and one column per catch group. In the `r common_name` example the catch is stratified by `state` and `geargroup`, leading to one column for each state-gear combination. Catches can be summarized in pounds of metric tons where the units should be explicitly passed to `getExpansion_2()` to convert the catches to pounds if necessary. 
+
+It is recommended to pass a pre-processed catch file to `formatCatch()` that includes data stratified by state, gear (e.g., expected name of geargroup), year (e.g., can use yr, year, landing_year, or sample_year), and catch values. Depending upon the species biological data in PacFIN can be present before the start of PacFIN catch records (1981), hence, catch amounts from historical reconstructions may be necessary to ensure proper expansion. Additionally, the stratification should always include areas (e.g., state or areas within states) even if using coastwide fleets in the model due to variable catch and sampling by areas.
+
+
+```{r catch, eval = FALSE}
+catch_formatted <- catch |>
+  formatCatch(
+    strat = c("state", "geargroup"),
+    valuename = "catch_mt"
+  )
+```
+
+## Biological data
+
+The biological data must first be cleaned using `cleanPacFIN()`. Here, we choose to remove the rows of data that should not be used in the assessment by choosing `CLEAN = TRUE` but you can change that to `FALSE` and all rows will be saved. We also create an additional column that matches records to the same stratification used for the catch data above for weighting.
+
+```{r biological-data, eval = FALSE}
+bds_cleaned <- cleanPacFIN(
+  Pdata = bds.pacfin,
+  keep_gears = used_gears,
+  CLEAN = TRUE,
+  keep_age_method = good_age_method,
+  keep_sample_type = good_samples,
+  keep_sample_method = good_methods,
+  keep_length_type = good_lengths,
+  keep_states = good_states,
+  spp = common_name
+) |>
+  dplyr::mutate(
+    stratification = paste(state, geargroup, sep = ".")
+  )
+```
+
+`cleanPacFIN()` filters out data and adds additional columns that are used in the data processing.  One of the columns added, `fleet`, is used to separate the data by model fleet. The default values in `fleet` are set to the gear names defined by `keep_gears` in `cleanPacFIN()`.  The values in this column should be modified the user, if necessary.
```

Comment:
Add "by" to "modified the user". Alternatively, the sentence could be revised to note that the user could either modify the fleet values coming out of `cleanPacFIN()` or modify the fleet values in the data frame created by `getComps()` (or maybe `writeComps()`).